nsxt/resource_nsxt_policy_lb_server_ssl_profile.go

```diff
@@ -0,0 +1,308 @@
+	model.LBServerSslProfile_CIPHERS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,
+	model.LBServerSslProfile_CIPHERS_ECDHE_RSA_WITH_AES_256_CBC_SHA384,
+	model.LBServerSslProfile_CIPHERS_RSA_WITH_AES_128_CBC_SHA256,
+	model.LBServerSslProfile_CIPHERS_RSA_WITH_AES_128_GCM_SHA256,
+	model.LBServerSslProfile_CIPHERS_RSA_WITH_AES_256_CBC_SHA256,
+	model.LBServerSslProfile_CIPHERS_RSA_WITH_AES_256_GCM_SHA384,
+	model.LBServerSslProfile_CIPHERS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
+	model.LBServerSslProfile_CIPHERS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,
+	model.LBServerSslProfile_CIPHERS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+	model.LBServerSslProfile_CIPHERS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384,
+	model.LBServerSslProfile_CIPHERS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+	model.LBServerSslProfile_CIPHERS_ECDH_ECDSA_WITH_AES_128_CBC_SHA,
+	model.LBServerSslProfile_CIPHERS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256,
+	model.LBServerSslProfile_CIPHERS_ECDH_ECDSA_WITH_AES_128_GCM_SHA256,
+	model.LBServerSslProfile_CIPHERS_ECDH_ECDSA_WITH_AES_256_CBC_SHA384,
+	model.LBServerSslProfile_CIPHERS_ECDH_ECDSA_WITH_AES_256_GCM_SHA384,
+	model.LBServerSslProfile_CIPHERS_ECDH_RSA_WITH_AES_128_CBC_SHA,
+	model.LBServerSslProfile_CIPHERS_ECDH_RSA_WITH_AES_128_CBC_SHA256,
+	model.LBServerSslProfile_CIPHERS_ECDH_RSA_WITH_AES_128_GCM_SHA256,
+	model.LBServerSslProfile_CIPHERS_ECDH_RSA_WITH_AES_256_CBC_SHA384,
+	model.LBServerSslProfile_CIPHERS_ECDH_RSA_WITH_AES_256_GCM_SHA384,
+}
+
+var lbServerSslProfileProtocolsValues = []string{
+	model.LBServerSslProfile_PROTOCOLS_SSL_V2,
+	model.LBServerSslProfile_PROTOCOLS_SSL_V3,
+	model.LBServerSslProfile_PROTOCOLS_TLS_V1,
+	model.LBServerSslProfile_PROTOCOLS_TLS_V1_1,
+	model.LBServerSslProfile_PROTOCOLS_TLS_V1_2,
+}
+
+var lbServerSslProfilePathExample = getMultitenancyPathExample("/infra/lb-server-ssl-profiles/[profile]")
+
+var lbServerSslProfileSchema = map[string]*metadata.ExtendedSchema{
+	"nsx_id":       metadata.GetExtendedSchema(getNsxIDSchema()),
+	"path":         metadata.GetExtendedSchema(getPathSchema()),
+	"display_name": metadata.GetExtendedSchema(getDisplayNameSchema()),
+	"description":  metadata.GetExtendedSchema(getDescriptionSchema()),
+	"revision":     metadata.GetExtendedSchema(getRevisionSchema()),
+	"tag":          metadata.GetExtendedSchema(getTagsSchema()),
+	"context":      metadata.GetExtendedSchema(getContextSchema(false, false, false)),
+	"cipher_group_label": {
+		Schema: schema.Schema{
+			Type:         schema.TypeString,
+			ValidateFunc: validation.StringInSlice(lbServerSslProfileCipherGroupLabelValues, false),
+			Required:     true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "string",
+			SdkFieldName: "CipherGroupLabel",
+		},
+	},
+	"ciphers": {
+		Schema: schema.Schema{
+			Type: schema.TypeList,
+			Elem: &metadata.ExtendedSchema{
+				Schema: schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringInSlice(lbServerSslProfileCiphersValues, false),
```

Comment:
The list of ciphers varies with NSX release. It might be safer to remove the validation function.
But as we have the same for client_ssl_ciphers, perhaps it does not matter.

For consistency, we can also remove validation for cipher groups, even if that list is pretty static

Reply:
That's a good point, we should update the list with every release (hopefully the process will be automatic)
I'm also wondering whether we should include the whole list in the docs..

Reply:
I don't think so

Reply:
I removed the validation for this resource. Will remove the same validation in other resources in a separate PR.